api_tests/nodes/views/test_node_addons.py

```diff
@@ -831,3 +829,166 @@ def test_settings_detail_incomplete_PUT_raises_error(self):
+
+
+class TestNodeForwardAddon(NodeUnmanageableAddonTestSuiteMixin, ApiAddonTestCase):
+    short_name = 'forward'
+
+    @property
+    def _mock_folder_info(self):
+        return {
+            'url': 'http://google.com',
+            'label': 'Gewgle'
+        }
+
+    def setUp(self):
+        super(TestNodeForwardAddon, self).setUp()
+        self.addon_type = 'OAUTH'
+        self.node_settings = self.node.get_or_add_addon(self.short_name, auth=self.auth)
+        self.node_settings.url = 'http://google.com'
+        self.node_settings.save()
+
+    ## Overrides
+
+    def test_settings_detail_GET_enabled(self):
+        res = self.app.get(
+            self.setting_detail_url,
+            auth=self.user.auth)
+
+        addon_data = res.json['data']['attributes']
+        assert_equal(self.node_settings.url, addon_data['url'])
+        assert_equal(self.node_settings.label, addon_data['label'])
+
+    def test_settings_detail_POST_enables(self):
+        self.node.delete_addon(self.short_name, auth=self.auth)
+        res = self.app.post_json_api(self.setting_detail_url, 
+            {'data': { 
+                'id': self.short_name,
+                'type': 'node_addons',
+                'attributes': {
+                    }
+                }
+            },
+            auth=self.user.auth)
+
+        addon_data = res.json['data']['attributes']
+        assert_equal(addon_data['url'], None)
+        assert_equal(addon_data['label'], None)
+
+        self.node.reload()
+        assert_not_equal(self.node.logs[-1].action, 'forward_url_changed')
+
+    def test_settings_detail_noncontrib_public_can_view(self):
+        self.node.set_privacy('public', auth=self.auth)
+        noncontrib = AuthUserFactory()
+        res = self.app.get(
+            self.setting_detail_url,
+            auth=noncontrib.auth)
+
+        assert_equal(res.status_code, 200)
+        addon_data = res.json['data']['attributes']
+        assert_equal(self.node_settings.url, addon_data['url'])
+        assert_equal(self.node_settings.label, addon_data['label'])
+
+    def test_settings_list_GET_enabled(self):
+        res = self.app.get(
+            self.setting_list_url,
+            auth=self.user.auth)
+
+        addon_data = self.get_response_for_addon(res)
+        assert_equal(self.node_settings.url, addon_data['url'])
+        assert_equal(self.node_settings.label, addon_data['label'])
+
+    def test_settings_list_noncontrib_public_can_view(self):
+        self.node.set_privacy('public', auth=self.auth)
+        noncontrib = AuthUserFactory()
+        res = self.app.get(
+            self.setting_list_url,
+            auth=noncontrib.auth)
+        addon_data = self.get_response_for_addon(res)
+
+        assert_equal(self.node_settings.url, addon_data['url'])
+        assert_equal(self.node_settings.label, addon_data['label'])
+
+    def test_settings_detail_PATCH_to_add_folder_without_auth_conflict(self):
+        pass
```

Comment:
Since this sort of thing is generally discouraged when we make tests, it's probably worth putting in a comment about it. I presume it's because these tests don't make sense for this addon but you need them for your testing suite?